lua/telescope/utils.lua

```diff
@@ -221,11 +221,13 @@ utils.diagnostics_to_tbl = function(opts)
   local buffer_diags = opts.get_all and vim.lsp.diagnostic.get_all()
     or { [current_buf] = vim.lsp.diagnostic.get(current_buf, opts.client_id) }
   for bufnr, diags in pairs(buffer_diags) do
-    for _, diag in ipairs(diags) do
-      -- workspace diagnostics may include empty tables for unused bufnr
-      if not vim.tbl_isempty(diag) then
-        if filter_diag_severity(opts, diag.severity) then
-          table.insert(items, preprocess_diag(diag, bufnr))
+    if not opts.only_cwd or (opts.only_cwd and string.find(vim.api.nvim_buf_get_name(bufnr), vim.loop.cwd(), 1, true) ) then
```

Comment:
What about a more general `opts.dir` that takes the folder path as an argument where you'd pass `vim.loop.cwd`? The slight advantages

1. Caches the folder out-of-the-box
2. Would flexibly restrict for the folder and its subfolders

Then you can simplify slightly in addition

```lua
if not opts.dir or string.find(vim.api.nvim_buf_get_name(bufnr), opts.dir, 1, true) then
```
 And add the `vim.loop.cwd()` to the docs as a general use-case for `opts.dir`.

Or generalizing even further (not sure if it's too complicated), but `opts.dir` could even be an `opts.filter_path` function that takes the buffer name in which you'd allow for more complex logic, i.e.

```lua
opts.filter_path = function(buf_name) return string.find(buf_name, vim.loop.cwd(), 1, true) end
...
if not opts.filter_path or opts.filter_path(vim.api.nvim_buf_get_name(bufnr)) then
```
`filter_path` could maybe have a better name :laughing: 

Thoughts?

Imo, the latter would be most flexible and instructive if a good example is added to the docs :) (Grep in telescope and look for `<pre>` to see how you could add that as a small code snippet).

Reply:
@Conni2461 thoughts?

Reply:
first ping me, i read it and then i read nvm :rofl: 

My take is, i hardly care about rust and i never run into the issue that i only needed a subdir for `workspace_diagnostics` if a only need a subset of diagnostics i can just do `document_diagnostics`.

So right now i would say `opts.only-cwd` is fine, if you think the other thing is more useful then feel free to implement it. You have 100% free hand with this :laughing:

Reply:
> Imo, the latter would be most flexible and instructive if a good example is added to the docs :) (Grep in telescope and look for <pre> to see how you could add that as a small code snippet).

Of course your option sounds way more complicated for endusers right now :laughing:

Reply:
I guess apart from rust `opts.filter_path` it can be interesting to eg to filter for subdirectories or parsing for imports of current file, but I won't use this myself. Then just merge this I suppose, since it should work.